src/app/shared/core/services/context.service.ts

```diff
@@ -12,20 +12,24 @@ import { IWebRequestService } from './web-request.interface';
  */
 @Injectable()
 export class ContextService implements IContextService {
-  public currentUser!: BehaviorSubject<User | undefined>;
+  public currentUser: BehaviorSubject<User | undefined> =
+    new BehaviorSubject<User | undefined>(undefined);
 
   constructor(
     private readonly webRequest: IWebRequestService,
     private readonly jsonConverter: IJsonConverterService,
   ) {
-    this.currentUser = new BehaviorSubject<User | undefined>(undefined);
   }
 
   public async isAuthenticated(): Promise<boolean> {
-    if (this.currentUser?.value) {
+    if (this.currentUser.value) {
       return true;
     }
+    await this.refreshUser();
```

Comment:
`isAuthenticated` in EstatePilot is not a promise in order to be a user in not async methods. 
Why is `refreshUser` needed here? 
As a golden rule, a query method cannot do any action (side effects)

Reply:
Still, an interrogatory method (`isSomething`) should not have side effects (`refreshUser`). We should follow Estate's behaviour to have a non promise `isAuthenticated()`. 
Please refresh the user `await this.refreshUser();` in the called of this method and leave `isAuthenticated` do only one job.